### PR TITLE
Add skew Hadamard matrix of order 1588

### PR DESCRIPTION
**Pull request by MatteoCati**
Added  the construction for spin type Goethals-Seidel difference family, described in the paper "Two classes of Hadamard matrices of Goethals-Seidel type" by D. Djokovic. This is done so that the skew Hadamard matrix of order 1588 is now present in SageMath.

### :memo: Checklist

- [X] The title is concise and informative.
- [X] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [X] I have created tests covering the changes.
- [X] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/combinat/designs/difference_family.py [hunk 1]
```diff
@@ -2193,6 +2193,316 @@ def generate_subset(indices, H):
...
+def _construct_gs_difference_family_from_full(S1, S2, mu):
+    r"""
+    Construct a spin type Goethals-Seidel difference family given the first two
+    sets.
+
+    This construction is described in [Djo2024]_. Given the first two sets
+    `S_1, S_2`, and the multiplier `\mu`, the last two sets are computed as
+    `S_3 = \mu S_2`, `S_4 =\mu S_3`.
+
+    The sets should contain elements of a group modulo `n`.
```
Comment (dimpase) [inline src/sage/combinat/designs/difference_family.py]: what does this mean? invertible elements in $\mathbb{Z}/n\mathbb{Z}$ ? Something else?

Comment (MatteoCati) [reply src/sage/combinat/designs/difference_family.py]: I've changed it to make it clearer.

**Review by dimpase — COMMENTED** [review]
replace "a group" with "the group" - integers mod n are essentially unique.

**Comment by dimpase** [comment]
please rebase

**Comment by dimpase** [comment]
otherwise, good

**Review by dimpase — APPROVED** [review]

**Pull request merged**